tfworker/definitions.py

```diff
@@ -79,9 +83,25 @@ def limited(self):
     def path(self):
         return self._path
 
+    @property
+    def provider_names(self):
+        """ Extract only the providers used by a definition """
+        result = set(self._providers.keys())
+        if self._tf_version_major >= 13:
+            version_path = os.path.join(self._target, "versions.tf")
+            if os.path.exists(version_path) and os.path.isfile(version_path):
```

Comment:
Is there a case where this exists and is not a file and the terraform will still work?

Reply:
good point. isfile is probably redundant.